Make: organize makefile defs

ldelossa opened the pull request:
house keeping task of organizing our Makefiles a bit neater.

this commit is step one a series of follow up commits for making our Makefiles more readable.

in this commit, various variables and helper functions are organized and segmented into section, hopefully making finding things a bit easier.

Signed-off-by: Louis DeLosSantos <louis.delos@isovalent.com>

```release-note
Segment Makefile defs into a series of sections.
```

joestringer commented on Makefile:
```diff
@@ -113,6 +103,7 @@ define generate_k8s_protobuf
 endef
 
 build: check-sources $(SUBDIRS) ## Builds all the components for Cilium by executing make in the respective sub directories.
+	echo $(GO_VERSION_FILE)
```
In general I use `make --quiet V=0` specifically to get the minimal amount of output, and in such cases I prefer not to see spurious messages in the output that don't relate to the current step in the build process.

I suggest dropping it:

```suggestion
```

Another alternative is to preface the line with `$(QUIET)`, which will hide the output in the silent mode like how I run it above.

ldelossa replied:
Okay, lets use $(QUIET) so that if someone does want to show the out, they can grep thru and remove that line.

joestringer commented on Makefile:
```diff
@@ -620,6 +611,7 @@ licenses-all: ## Generate file with all the License from dependencies.
 	@$(GO) run ./tools/licensegen > LICENSE.all || ( rm -f LICENSE.all ; false )
 
 check-go-version: ## Check locally install Go version against required Go version.
+	echo $(ROOT_DIR)
```
```suggestion
```

joestringer reviewed (commented):
First couple of things I noticed. I didn't look super closely at the rest given the number of lines changing, but I plan to take another look. I presume it's mostly just shuffling code from one place to another anyway.

ldelossa committed:
house keeping task of organizing our Makefiles a bit neater.

this commit is step one a series of follow up commits for making our Makefiles more readable.

in this commit, various variables and helper functions are organized and segmented into section, hopefully making finding things a bit easier.

Signed-off-by: Louis DeLosSantos <louis.delos@isovalent.com>
